bulk-export/bulk-export-service/service.bal

```diff
@@ -1,33 +1,88 @@
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
 import ballerina/http;
+import ballerinax/health.fhir.r4;
 import ballerina/log;
+import ballerina/uuid;
+import ballerina/time;
+import ballerina/io;
 
-configurable int port = 9091;
+configurable SearchServerConfig searchServerConfig = ?;
+configurable BulkExportServerConfig exportServiceConfig = ?;
 
-# initialize source system endpoint here
-
-// Since 'status' and 'bulk-file' endpoints are not part of the FHIR standard, we define them here.
-// When deploying the services, both FHIR and http services must have the same host.
-service / on new http:Listener(port) {
-
-    // download exported files.
-    isolated resource function get api/export/[string taskId]/bulkfiles/[string fileName]() returns @http:Payload{mediaType: "application/json"} http:Response {
-        // read ndjson from file system and return
-
-        log:printInfo(string `taskId: ${taskId}, fileName: ${fileName}`);
+service / on new http:Listener(exportServiceConfig.port) {
+    isolated resource function get fhir/r4/Patient/export () returns r4:OperationOutcome|r4:FHIRError {
+        string exportTaskId = uuid:createType1AsString();
+        error? executionResult = executeJob(exportTaskId, searchServerConfig, exportServiceConfig);
+        if executionResult is error {
+            log:printError("Error occurred: ", executionResult);
+            return r4:createFHIRError("Server Error", r4:ERROR, r4:PROCESSING, httpStatusCode = http:STATUS_INTERNAL_SERVER_ERROR);
+        }
+        addExportTasktoMemory(exportTaskId, time:utcNow());
+        
+        return createOpereationOutcome("information", "processing",
+         "Your request has been accepted. You can check its status at " + exportServiceConfig.baseUrl + "/fhir/bulkstatus/" + exportTaskId);
+    }
 
-        string filePath = "resources/" + fileName;
+    isolated resource function get fhir/bulkstatus/[string exportTaskId] () returns json|r4:FHIRError|http:Response {
+        ExportTask|error exportTask = getExportTaskFromMemory(exportTaskId);
 
-        http:Response response = new http:Response();
-        response.setFileAsPayload(filePath);
-        return response;
+        if exportTask is error {
+            http:Response response = new;
+            response.statusCode = http:STATUS_BAD_REQUEST;
+            response.setPayload(exportTask.message());
+            return response;
+        }
+        
+        if exportTask.lastStatus == "in-progress" {
+            return;
```

Comment:
Don't we need to set the http status as 202 for this case?

Reply:
202 is responded by default. No need to explicitly mention

Reply:
ack